Framework/CurveFitting/src/Functions/CrystalFieldMultiSpectrum.cpp

```diff
@@ -214,6 +267,60 @@ API::IFunction_sptr CrystalFieldMultiSpectrum::buildSpectrum(
   return IFunction_sptr(spectrum);
 }
 
+API::IFunction_sptr CrystalFieldMultiSpectrum::buildPhysprop(
+    int nre, const DoubleFortranVector &en, const ComplexFortranMatrix &wf,
+    const ComplexFortranMatrix &ham, double temperature, size_t iSpec) const {
+  switch (m_physprops[iSpec]) {
+  case 1: { // Heat capacity
+    auto spectrum = new CrystalFieldHeatCapacity;
+    spectrum->set_eigensystem(en, wf, nre);
+    return IFunction_sptr(spectrum);
+  }
+  case 2: { // Susceptibility
+    auto spectrum = new CrystalFieldSusceptibility;
+    spectrum->set_eigensystem(en, wf, nre);
+    auto suffix = std::to_string(iSpec);
+    auto hdir = getAttribute("Hdir" + suffix).asVector();
+    spectrum->setAttribute("Hdir", Attribute(hdir));
```

Comment:
There is no need to convert attributes to their value types, they can be copied directly.